### PR TITLE
Update run.sh

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 163a3e44d38b249155f4c056e6388b0cffe26153 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59280/163a3e44d38b249155f4c056e6388b0cffe26153/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59280/163a3e44d38b249155f4c056e6388b0cffe26153/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by kssenii** [comment]
> AST fuzzer (tsan)

https://github.com/ClickHouse/ClickHouse/issues/59310

> Upgrade check (asan) 

https://github.com/ClickHouse/ClickHouse/issues/57893

File: docker/test/stateless/run.sh [hunk 1]
```diff
@@ -246,16 +246,19 @@ clickhouse-client -q "system flush logs" ||:
...
 
 # Try to get logs while server is running
-successfuly_saved=0
+failed_to_save_logs=0
 for table in query_log zookeeper_log trace_log transactions_info_log metric_log
 do
-    clickhouse-client -q "select * from system.$table format TSVWithNamesAndTypes" | zstd --threads=0 > /test_output/$table.tsv.zst
-    successfuly_saved=$?
+    err=$( { clickhouse-client -q "select * from system.$table format TSVWithNamesAndTypes" | zstd --threads=0 > /test_output/$table.tsv.zst; } 2>&1 )
+    echo $err
```
Comment (serxa) [inline docker/test/stateless/run.sh:253]: If we want to keep the previous behavior it should be redirected to stderr (and the same below)

**Review by serxa — COMMENTED** [review]
If we want just to check exit code of clickhouse as well as zstd, maybe just use `
set -o pipefail`? At least for this part of the script.

File: docker/test/stateless/run.sh (hunk 1 above)
Comment (kssenii) [reply docker/test/stateless/run.sh]: I think it does not matter where it is redirected, it matters that we see it in `run.log`.

Comment (kssenii) [reply docker/test/stateless/run.sh]: > If we want just to check exit code of clickhouse as well as zstd, maybe just use  set -o pipefail? At least for this part of the script.

Is it ok if I set it in a separate follow-up PR? I want this fix for run.sh asap to debug another issue...

Comment (serxa) [reply docker/test/stateless/run.sh]: Sure, maybe it is not needed at all, if this works

**Review by serxa — APPROVED** [review]

**Pull request merged**